rest/admin_api.go

```diff
@@ -173,22 +177,34 @@ func (h *handler) handleCreateDB() error {
 	return base.HTTPErrorf(http.StatusCreated, "created")
 }
 
-// getAuthScopeHandleCreateDB is used in the router to supply an auth scope for the admin api auth. Takes the JSON body
-// from the payload, pulls out bucket and returns this as the auth scope.
-func getAuthScopeHandleCreateDB(ctx context.Context, bodyJSON []byte) (string, error) {
-	var body struct {
+// getAuthScopeHandleCreateDB determines the auth scope for a PUT /{db}/ request.
+// This is a special case because we don't have a database initialized yet, so we need to infer the bucket from db config.
+func getAuthScopeHandleCreateDB(ctx context.Context, h *handler) (bucketName string, err error) {
+
+	// grab a copy of the request body and restore body buffer for later handlers
+	bodyJSON, err := h.readBody()
+	if err != nil {
+		return "", base.HTTPErrorf(http.StatusInternalServerError, "Unable to read body: %v", err)
+	}
+	// mark the body as already read to avoid double-counting bytes for stats once it gets read again
+	h.requestBody.bodyRead = true
+	h.requestBody.reader = io.NopCloser(bytes.NewReader(bodyJSON))
+
+	var dbConfigBody struct {
 		Bucket string `json:"bucket"`
 	}
 	reader := bytes.NewReader(bodyJSON)
-	err := DecodeAndSanitiseConfig(ctx, reader, &body, false)
+	err = DecodeAndSanitiseConfig(ctx, reader, &dbConfigBody, false)
 	if err != nil {
 		return "", err
 	}
-	if body.Bucket == "" {
-		return "", nil
+
+	if dbConfigBody.Bucket == "" {
+		// imply bucket name from db name in path if not in body
+		return h.PathVar("newdb"), nil
```

Comment:
Is it worth checking if `newdb` is empty string here? I don't think it could but it might be better to error rather than try to figure it out later if we re-use getAuthScopeHandleCreateDB